client/components/pie-chart/index.js

```diff
@@ -0,0 +1,114 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { pie as d3Pie, arc as d3Arc } from 'd3-shape';
+import { isEqualWith, sortBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import LegendItem from './legend-item';
+
+const NUM_COLOR_SECTIONS = 3;
+const SVG_SIZE = 300;
+
+class PieChart extends Component {
+	static propTypes = {
+		data: PropTypes.arrayOf( PropTypes.shape( {
+			description: PropTypes.string,
+			name: PropTypes.string.isRequired,
+			value: PropTypes.number.isRequired,
+		} ) ).isRequired,
+		title: PropTypes.string,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = this.processData( props.data );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const customizer = ( datum, otherDatum ) => datum.value === otherDatum.value;
```

Comment:
Since the chart renders from the state, we actually cannot get away with only checking the value, because it will prevent changes to the name & descriptions being rendered.

Reply:
Good catch, I missed that in the review.